core/src/main/java/org/mule/runtime/core/api/transaction/xa/XaTransaction.java

```diff
@@ -387,7 +387,7 @@ public Object getResource() {
 
     @Override
     public int hashCode() {
-      return System.identityHashCode(resourceFactory);
+      return Objects.hashCode(resourceFactory);
```

Comment:
because the `resourceFactory` will a `Reference` referencing a `ConnectionProvider`, due that different `Reference` instances pointing to the same `ConnectionProvider` could appear, the `identityHashCode` is useless